### PR TITLE
Phkuo/theme generator

### PR DESCRIPTION
**Pull request by phkuo**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

This creates the Theme Generator page, a tool to help users to easily automatically generate a theme based on a few color choices. It also includes hidden functionality to generate a theme from an image. (The user will need to enlist and modify the code to use their own subscription key.)

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by phkuo** [comment]
todo:
- add powershell formatted output
- add to external-facing website (once I figure out how to actually verify a change like this)

File: packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx [hunk 1]
```diff
@@ -11,131 +11,131 @@ const ThemeCodeExample = require('!raw-loader!office-ui-fabric-react/src/compone
 
 export class ThemePage extends React.Component<any, any> {
```
Comment (phkuo) [inline packages/office-ui-fabric-react/src/components/Theme/ThemePage.tsx]: I think we can just completely remove this page now?

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.Props.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+     */
+    onColorChanged?: (color: string) => void;
 
-  /**
-   * The setting of whether hide alpha control slider.
-   */
-  alphaSliderHidden?: boolean;
+    /**
+     * The setting of whether hide alpha control slider.
+     */
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.Props.ts]: tabs are messed up.

**Comment by phkuo** [comment]
image

File: packages/utilities/src/object.ts [hunk 3]
```diff
@@ -85,3 +85,16 @@ export function getId(prefix?: string): string {
...
+ * The callback takes as a first parameter the string that represents the name of the entry, and the second parameter is the
+ * value of that entry, which is the value you'd normally use when using the enum (usually a number).
+ * */
+export function mapEnumByName<T>(theEnum: any, callback: (name?: string, value?: any) => T | undefined): T[] | undefined {
+  // map<any> to satisfy compiler since it doesn't realize we strip out undefineds in the .filter() call
+  return Object.keys(theEnum).map<any>((p: string) => { // map on each property name as a string
+    if (String(Number(p)) !== p) { // if the property is not just a number (because enums in TypeScript will map both ways)
+      return callback(p, theEnum[p]);
+    }
+  }).filter((v: any) => !!v); // only return elements with values
```
Comment (dzearing) [inline packages/utilities/src/object.ts]: no unit test?

**Pull request merged**